Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -429,28 +429,14 @@ void WebFrameProxy::prepareForProvisionalLoadInProcess(WebProcessProxy& process,
     if (isMainFrame())
         return completionHandler();
 
-    if (m_provisionalFrame && m_provisionalFrame->process().processID() == process.processID())
-        return completionHandler();
-
-    if (process.coreProcessIdentifier() == this->process().coreProcessIdentifier()) {
-        m_provisionalFrame = nullptr;
-        return completionHandler();
-    }
-
     RegistrableDomain navigationDomain(navigation.currentRequest().url());
-    // addAllowedFirstPartyForCookies can be sync, but we need completionHander to be invoked after this function.
-    auto aggregator = CallbackAggregator::create(WTFMove(completionHandler));
-    if (!m_provisionalFrame || m_provisionalFrame->process().coreProcessIdentifier() != process.coreProcessIdentifier() || navigation.currentRequestIsCrossSiteRedirect()) {
-        RefPtr page = m_page.get();
-        // FIXME: Main resource (of main or subframe) request redirects should go straight from the network to UI process so we don't need to make the processes for each domain in a redirect chain. <rdar://116202119>
-        RegistrableDomain mainFrameDomain(page->mainFrame()->url());
-
-        m_provisionalFrame = ProvisionalFrameProxy::create(*this, group.ensureProcessForDomain(navigationDomain, process, page->preferences()));
-        page->websiteDataStore().protectedNetworkProcess()->addAllowedFirstPartyForCookies(process, mainFrameDomain, LoadedWebArchive::No, [aggregator] { });
-    }
+    RefPtr page = m_page.get();
+    // FIXME: Main resource (of main or subframe) request redirects should go straight from the network to UI process so we don't need to make the processes for each domain in a redirect chain. <rdar://116202119>
+    RegistrableDomain mainFrameDomain(page->mainFrame()->url());
 
-    if (this->process().processID() != process.processID())
-        process.send(Messages::WebPage::CreateProvisionalFrame({ m_layerHostingContextIdentifier }, frameID()), page()->webPageIDInProcess(process));
```

Comment:
nit - you can remove `#include "ProvisionalFrameCreationParameters.h"` in this file now.